minimal/core_01_minimal.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * @license
+ * Code City: Minimal database.
+ *
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Minimal database for Code City.
+ * @author fraser@google.com (Neil Fraser)
+ */
+
+var user = null;
+var $ = {};
+
+// System object: $.system
+$.system = {};
+$.system.log = new '$.system.log';
+$.system.checkpoint = new '$.system.checkpoint';
+$.system.shutdown = new '$.system.shutdown';
+$.system.connectionWrite = new 'connectionWrite';
+$.system.connectionClose = new 'connectionClose';
+
+
+// Prototype physical object: $.physical
+$.physical = {};
+$.physical.name = 'Prototype physical object';
```

Comment:
Naming convention: "Array prototype" or "Prototype array"?  I think the former is more usual, so hence this would be "Physical object prototype" (which also puts the more distinguishing part of the name at the beginning).